packages/core/src/checkout-buttons/strategies/googlepay/googlepay-button-strategy.ts

```diff
@@ -1,29 +1,41 @@
 import { FormPoster } from '@bigcommerce/form-poster';
 
 import { CheckoutButtonMethodType } from '../';
+import { Cart, CartRequestSender } from '../../../cart';
+import { BuyNowCartCreationError } from '../../../cart/errors';
 import { CheckoutActionCreator, CheckoutStore } from '../../../checkout';
-import { InvalidArgumentError, NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
+import {
+    InvalidArgumentError,
+    MissingDataError, MissingDataErrorType,
+    NotInitializedError,
+    NotInitializedErrorType
+} from '../../../common/error/errors';
 import { SDK_VERSION_HEADERS } from '../../../common/http-request';
 import { bindDecorator as bind } from '../../../common/utility';
-import { GooglePayPaymentProcessor } from '../../../payment/strategies/googlepay';
+import { GooglePayPaymentProcessor, } from '../../../payment/strategies/googlepay';
 import { getShippableItemsCount } from '../../../shipping';
 import { CheckoutButtonInitializeOptions } from '../../checkout-button-options';
 import CheckoutButtonStrategy from '../checkout-button-strategy';
 
 import { GooglePayButtonInitializeOptions } from './googlepay-button-options';
+
+type BuyNowInitializeOptions = Pick<GooglePayButtonInitializeOptions, 'buyNowInitializeOptions'>
+
 export default class GooglePayButtonStrategy implements CheckoutButtonStrategy {
     private _methodId?: string;
     private _walletButton?: HTMLElement;
+    private _buyNowCart?: Cart;
 
     constructor(
         private _store: CheckoutStore,
         private _formPoster: FormPoster,
         private _checkoutActionCreator: CheckoutActionCreator,
-        private _googlePayPaymentProcessor: GooglePayPaymentProcessor
+        private _googlePayPaymentProcessor: GooglePayPaymentProcessor,
+        private _cartRequestSender?: CartRequestSender,
```

Comment:
Why is this an optional?

Reply:
@animesh1987 
Because **GooglePayButtonStrategy** is used in many places where we don't need this instance
We only need it for **googlepaybraintree**

But we can work around this if we create an instance here and set it to a private variable as an alternative

Reply:
I think it's better to leave it as is
Alternative option has unnecessary steps

Reply:
Hey @bc-nick we might need to update other methods to provide the same feature set, so feel better to make it required?

Reply:
@animesh1987 okay. i have made it required